angularSTEP/src/app/recipe-page/recipe-page.component.ts

```diff
@@ -13,17 +19,100 @@ import {RecipeConverter} from '../recipe-converter';
 export class RecipePageComponent {
   id: string | null = this.route.snapshot.paramMap.get('id');
   pageRecipe!: Recipe;
+  user!: User;
 
-  constructor(private db: AngularFirestore,
-              private route: ActivatedRoute,) {
-                this.setRecipeData();
+  constructor(
+    private db: AngularFirestore,
+    private route: ActivatedRoute,
+    private fAuth: AngularFireAuth,
+    private dialog: MatDialog,
+    private snackBar: MatSnackBar,
+  ) {
+    this.setRecipeData();
+    this.fAuth.currentUser.then(user => {
+      if (user) {
+        this.setUserData(user.uid);
+      }
+    });
   }
 
   async setRecipeData() {
-    const postRecipe = await this.db.doc('/recipes/' + this.id + '/').ref.withConverter(new RecipeConverter().recipeConverter).get();
+    const postRecipe = await this.db.doc('/recipes/' + this.id + '/')
+      .ref.withConverter(new RecipeConverter().recipeConverter).get();
     const recipeClassData = postRecipe.data();
     if (recipeClassData) {
       this.pageRecipe = recipeClassData;
     }
   } 
+
+  async setUserData(uid: string) {
+    const postUser = await this.db.doc('/users/' + uid + '/')
+      .ref.withConverter(new Converter().userConverter).get();
+    if (postUser.data()) {
+      this.user = postUser.data()!;
+    } 
+  }
+
+  objToMap(obj: any): Map<string, number> {
+    const mp = new Map();
+    Object.keys(obj).forEach(k => {mp.set(k, obj[k])});
+    return mp;
+  }
+
+
+  addItem(item: string) {
+    var currentShoppingList: Map<string, number> = this.objToMap(this.user.shoppingList);
+
+    const dialogConfig = new MatDialogConfig();
+
+    dialogConfig.disableClose = true;
+    dialogConfig.autoFocus = true;
+    dialogConfig.height = 'auto';
+    dialogConfig.width = 'auto';
+    dialogConfig.data = {
+      selectedItem: item,
+      add: true,
+      remove: false,
+      max: 0,
+    };
+
+    const dialogRef = this.dialog.open(ItemDialogComponent, dialogConfig);
+
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        if (currentShoppingList.has(item)) {
+          var currentValue = currentShoppingList.get(item);
+          if (currentValue) {
+            currentShoppingList.set(item, (currentValue + result));
+          }
+        } else {
+          currentShoppingList.set(item, result);
+        }  
+
+        let objectCurrentShoppingList = Array.from(currentShoppingList).reduce((objectCurrentShoppingList, [key, value]) => (
+          Object.assign(objectCurrentShoppingList, { [key]: value }) 
+        ), {});
+    
+        this.fAuth.currentUser.then(user => {
+          if (user) {
+            this.db.collection('users')
+            .doc(this.user.uid)
+            .ref.withConverter(new Converter().userConverter)
+            .update({shoppingList: objectCurrentShoppingList})
+            .then(() => {
+              this.setUserData(user.uid);
+              let snackBarRef = this.snackBar.open('Item Added!', undefined, {
```

Comment:
It seems like `snackBarRef` isn't needed/used for anything, why declare+assign to it if simply calling `this.snackBar.open()` and ignoring the returned value does the same thing?

Reply:
I was following the documentation at https://material.angular.io/components/snack-bar/overview, but you're right that there isn't a reason to declare and assign it for the capacity that it's being used in, so I'll change that now as well.